orc/orcarm.h

```diff
@@ -400,6 +400,350 @@ ORC_API void orc_arm_emit_lg (OrcCompiler *p, OrcArm64RegBits bits, OrcArm64DP o
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_UXTH,Rd,Rn,0,imm)
+#define orc_arm64_emit_ands_uxtw(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_UXTW,Rd,Rn,0,imm)
+#define orc_arm64_emit_ands_uxtx(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_UXTX,Rd,Rn,0,imm)
+#define orc_arm64_emit_ands_sxtb(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_SXTB,Rd,Rn,0,imm)
+#define orc_arm64_emit_ands_sxth(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_SXTH,Rd,Rn,0,imm)
+#define orc_arm64_emit_ands_sxtw(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_SXTW,Rd,Rn,0,imm)
+#define orc_arm64_emit_ands_sxtx(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_SXTX,Rd,Rn,0,imm)
+
+/** ORC_ARM64_DP_TST (alias os ANDS) */
+/** ORC_ARM64_TYPE_IMM */
+#define orc_arm64_emit_tst_i(p,bits,Rd,Rn,imm) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_IMM,0,0,Rn,0,imm)
+/** ORC_ARM64_TYPE_REG */
+#define orc_arm64_emit_tst_r(p,bits,Rd,Rn,Rm) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_REG,0,0,Rn,0,imm)
+#define orc_arm64_emit_tst_lsl(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_REG,ORC_ARM_LSL,0,Rn,0,imm)
+#define orc_arm64_emit_tst_asr(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_REG,ORC_ARM_ASR,0,Rn,0,imm)
+#define orc_arm64_emit_tst_ror(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_REG,ORC_ARM_ROR,0,Rn,0,imm)
+/** ORC_ARM64_TYPE_EXT */
+#define orc_arm64_emit_tst_uxtb(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_UXTB,0,Rn,0,imm)
+#define orc_arm64_emit_tst_uxth(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_UXTH,0,Rn,0,imm)
+#define orc_arm64_emit_tst_uxtw(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_UXTW,0,Rn,0,imm)
+#define orc_arm64_emit_tst_uxtx(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_UXTX,0,Rn,0,imm)
+#define orc_arm64_emit_tst_sxtb(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_SXTB,0,Rn,0,imm)
+#define orc_arm64_emit_tst_sxth(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_SXTH,0,Rn,0,imm)
+#define orc_arm64_emit_tst_sxtw(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_SXTW,0,Rn,0,imm)
+#define orc_arm64_emit_tst_sxtx(p,bits,Rd,Rn,Rm,val) \
+  orc_arm64_emit_lg(p,bits,ORC_ARM64_DP_ANDS,ORC_ARM64_TYPE_EXT,ORC_ARM64_SXTX,0,Rn,0,imm)
+
```

Comment:
Whooah... you are going to have a real hard time making test cases later :P

Reply:
Yes... I will think about an easy way to verify these macros.